File: 백준/Silver/13305. 주유소/주유소.py
import sys
input= sys.stdin.readline

city=int(input())
dist=list(map(int,input().split()))
price=list(map(int,input().split()))
total=0
j=0
for i in range (0,city):
    min_price=min(price[0:i+1])
    if j==len(dist): break
    else:
        total=total+min_price*dist[j]
        j+=1

print(total)

File: 백준/Silver/13305. 주유소/주유소.js
let input = require('fs').readFileSync('dev/stdin').toString().trim().split('\n');
let N = Number(input[0]);
let distance = input[1].split(' ').map(BigInt);
let gasStation = input[2].split(' ').map(BigInt);

let totalCost = 0n;
let minPrice = gasStation[0];

for (let i = 0; i < N - 1; i++) {
  if (gasStation[i] < minPrice) {
    minPrice = gasStation[i];
  }
  totalCost += minPrice * distance[i];
}

console.log(totalCost.toString());
